### ISSUE TITLE
9$: 'declare global' requires an 'export' somewhere in the file

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1091]
Legal:
```typescript
declare global { }
export default 42;
```

Legal:
```typescript
declare global { }
import fs from 'node:fs';
```

Illegal:
```typescript
declare global { }
```

TypeScript gives a bad diagnostic. We should give diagnostic which is helpful. Something like:
> 'declare global' is only allowed in a TypeScript file containing at least one 'export' or 'import'

**Comment by strager** [comment]
Relatedly, 'declare module "name"' requires *no* 'export' or 'import' anywhere in the file.
